lib/posix-sysinfo: Provide _SC_GETPW_R_SIZE_MAX

andreittr opened the pull request:
### Description of changes

Calling code expects `sysconf(_SC_GETPW_R_SIZE_MAX)` to be either -1 or a non-zero value for an initial buffer size. The previous default value of 0 is outside of specifications and can cause unintended behavior. This change makes `sysconf(_SC_GETPW_R_SIZE_MAX)` return -1.

See `man 3 getpwuid` for the specifications for `sysconf(_SC_GETPW_R_SIZE_MAX)`.

### Prerequisite checklist

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [x] Updated relevant documentation.

### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): N/A

### Additional configuration

`CONFIG_LIBPOSIX_USER=y` and `CONFIG_LIBPOSIX_SYSINFO=y`.

andreittr committed:
lib/posix-sysinfo: Provide _SC_GETPW_R_SIZE_MAX

Calling code expects sysconf(_SC_GETPW_R_SIZE_MAX) to be either -1 or a
non-zero value for an initial buffer size. The previous default value of
0 is outside of specifications and can cause unintended behavior.
This change makes sysconf(_SC_GETPW_R_SIZE_MAX) return -1.

Signed-off-by: Andrei Tatar <andrei@unikraft.io>

StefanJum reviewed (changes requested):
@andreittr we should define `_SC_GETPW_R_SIZE_MAX` in [`nolibc/include/unistd.h`](https://github.com/unikraft/unikraft/blob/staging/lib/nolibc/include/unistd.h#L52-L60), otherwise the build fails when using `nolibc`.

andreittr commented:
Update: added define of _SC_GETPW_R_SIZE_MAX to nolibc.

StefanJum reviewed (approved):
Looks good now. I'll add the tag after @John-Ted and @razvand take a look.

razvand commented:
@John-Ted , please take a look, so I can then approve it and have it merged.

John-Ted commented:
I'm trying to compile [the helloworld app](https://github.com/unikraft/app-helloworld) to print the `sysconf` value, but I cannot get it to compile. I have selected the 2 options in the config, and I included `unistd.h`. I get the following errors:
```
  CC      apphelloworld: main.o
/home/teodor/Documents/ac/unikraft/pr_reviews/rev1/apps/app-helloworld/main.c: In function ‘main’:
/home/teodor/Documents/ac/unikraft/pr_reviews/rev1/apps/app-helloworld/main.c:34:37: warning: implicit declaration of function ‘sysconf’; did you mean ‘sscanf’? [-Wimplicit-function-declaration]
   34 |         printf("Hello world %d!\n", sysconf(_SC_GETPW_R_SIZE_MAX));
      |                                     ^~~~~~~
      |                                     sscanf
/home/teodor/Documents/ac/unikraft/pr_reviews/rev1/apps/app-helloworld/main.c:34:45: error: ‘_SC_GETPW_R_SIZE_MAX’ undeclared (first use in this function)
   34 |         printf("Hello world %d!\n", sysconf(_SC_GETPW_R_SIZE_MAX));
      |                                             ^~~~~~~~~~~~~~~~~~~~
/home/teodor/Documents/ac/unikraft/pr_reviews/rev1/apps/app-helloworld/main.c:34:45: note: each undeclared identifier is reported only once for each function it appears in
make[3]: *** [/home/teodor/Documents/ac/unikraft/pr_reviews/rev1/apps/app-helloworld/.unikraft/unikraft/support/build/Makefile.build:27: /home/teodor/Documents/ac/unikraft/pr_reviews/rev1/apps/app-helloworld/build/apphelloworld/main.o] Error 1
```
[config.txt](https://github.com/unikraft/unikraft/files/12152823/config.txt)
[main.c.txt](https://github.com/unikraft/unikraft/files/12152824/main.c.txt)

andreittr commented:
> I'm trying to compile [the helloworld app](https://github.com/unikraft/app-helloworld) to print the `sysconf` value, but I cannot get it to compile. I have selected the 2 options in the config, and I included `unistd.h`. I get the following errors:

Whoops, seems `sysconf` is not declared in nolibc's `unistd.h`; will push a new commit with a fix.

andreittr committed:
lib/nolibc: Provide declaration of sysconf

This change adds a missing declaration of `sysconf()` in the unistd.h
header provided by nolibc, complementing the _SC_* constants.

Signed-off-by: Andrei Tatar <andrei@unikraft.io>

John-Ted reviewed (approved):
Everything works fine now. Thanks!

Reviewed-by: Ioan-Teodor Teugea <ioan_teodor.teugea@stud.acs.upb.ro>

razvand reviewed (approved):
Approved-by: Razvan Deaconescu <razvand@unikraft.io>
